libraries/cmd/unit_generic.cpp

```diff
@@ -170,24 +172,65 @@ bool Unit::InRange(const Unit *target, double &mm, bool cone, bool cap, bool loc
     return true;
 }
 
+
+
+// two convenience functions
+Unit* CheckNullAndCastToUnit(ComponentsManager* manager) {
```

Comment:
IIRC, vega_dynamic_cast_ptr (and the const variant) were intended to handle nulls, as well. If they aren't handling nulls adequately, maybe we should look at how to fix that, instead of introducing two new functions that barely do anything.

Reply:
Huh. I was actually automatically using your functions without thinking. 
I've had crashes when trying to cast a null *_ComponentManager_ to *_Unit_.
I think _dynamic_cast_ is enough here.

Reply:
Perhaps `dynamic_cast` is enough. However, if we can encapsulate some common logic surrounding the cast, such as throwing an exception/aborting the game if the cast fails, then I think that might be worthwhile. That was the idea behind `vega_dynamic_cast_ptr` initially.

Reply:
Except I don't want to throw an exception and I never will.
The function checks for null and returns null in that case.
I should be able to return null Unit*.
The issue here is that if you try and dynamic cast null it will fail.

Reply:
Ultimately, a failing cast is a "shouldn't happen." If it happens, we should abort the game as gracefully as possible, offering the user a way to submit a bug report. That requires some common logic. How else are you going to do it?

Reply:
Looking at it with fresh eyes and rereading your comments, I think my original code was actually the best.
There are three separate scenarios here:
1. `manager` refers to a `Unit` and gets cast successfully.
2. `manager` is **null** and if cast, would crash the game. We should check this and return **null**.
3. `manager` isn't **null** but isn't an instance of `Unit`. This would be caught by `vega_dynamic_cast_ptr`.

Reply:
OK. I can go with your original code.

Reply:
I think the three scenarios you identified are correct and make sense. Thanks.